model/event/event_registry.go

```diff
@@ -68,9 +76,54 @@ func (r *eventRegistry) IsSubscribable(resourceType, eventType string) bool {
 		EventType:    eventType,
 	}
 
-	allowSubs := r.isSubscribable[e]
+	return r.isSubscribable[e]
+}
+
+// RegisterExtraData allows a pair of (resourceType, trigger) to deserialize
```

Comment:
I think we could name this more clearly. I'm not sure what "extra" means in this context. 

Should we make the methods on the event registry private, and then have public wrapper methods that access the global instance?

Reply:
> I think we could name this more clearly. I'm not sure what "extra" means in this context.

An alternative would be to add a comment here that explains what and why in addition to how. It's useful to know that this deserializes data from the ExtraData field, but even more useful to know what that is and why we use it.

Reply:
I'm partial to the comment. There is also no need to access the registry outside of this package.

Reply:
Could you add a sentence that explains the function at a higher level of abstraction? The current text is fine as is. It's additional clarification that I'm after.